src/ice_shelf/MOM_ice_shelf.F90

```diff
@@ -1729,9 +1939,23 @@ subroutine update_shelf_mass(G, US, CS, ISS, Time)
 
   ! local variables
   integer :: i, j, is, ie, js, je
+  real, allocatable, dimension(:,:) :: tmp2d ! Temporary array for storing ice shelf input data
+
   is = G%isc ; ie = G%iec ; js = G%jsc ; je = G%jec
 
-  call time_interp_external(CS%id_read_mass, Time, ISS%mass_shelf)
+
+  if (CS%rotate_index) then
+     allocate(tmp2d(CS%Grid_in%isc:CS%Grid_in%iec,CS%Grid_in%jsc:CS%Grid_in%jec));tmp2d(:,:)=0.0
+  else
```

Comment:
This line should end with ` ; tmp2(:,:) = 0.0` like the line above?

Reply:
This allocation, call to `time_interp_external`, and rotation could be replaced with `rotated_time_interp_external`, see `MOM_horizontal_regridding.F90`.